src/main/native/GCM.c

```diff
@@ -41,12 +41,41 @@ typedef struct {
 #define THREAD_LOCAL __thread
 #endif
 
-int  GCM_InitForUpdateDecrypt_core ();
-int  GCM_UpdForUpdateDecrypt_core ();
-int  GCM_FinalForUpdateDecrypt_core ();
-int  GCM_InitForUpdateEncrypt_core ();
-int  GCM_UpdForUpdateEncrypt_core ();
-int  GCM_FinalForUpdateEncrypt_core ();
+int GCM_InitForUpdateDecrypt_core(JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```

Comment:
Would it make sense to just move the actual methods up here, instead of declaring them and implementing them later?

Reply:
I did move all the functions higher in the file. The empty prototypes can still be removed as those prototypes with no arguments do not represent what the method really takes as arguments. The prototypes that i created earlier are now also removed and not needed once the methods were moved.